apprise/Apprise.py

```diff
@@ -319,16 +320,122 @@ def notify(self, body, title='', notify_type=NotifyType.INFO,
         such as turning a \n into an actual new line, etc.
         """
 
-        if len(self) == 0:
-            # Nothing to notify
+        if ASYNCIO_SUPPORT:
+            return py3compat.asyncio.tosync(
+                self.async_notify(
+                    body, title,
+                    notify_type=notify_type, body_format=body_format,
+                    tag=tag, attach=attach,
+                    interpret_escapes=interpret_escapes,
+                )
+            )
+
+        else:
+            try:
+                results = peekable(
```

Comment:
Not sure if we really need `peekable` here.  Is there a way to accomplish what you want here without the need for the additional Python requirement?

Reply:
My preference would be to continue using more_itertools. It's a small, pure Python library that doesn't depend on anything else. In addition, with f4387d116f819cb08338b753b148047eeefef7c6, I've had to borrow another function from it.

But if you want to drop this dependency, the need for it stems entirely from my use of lazy generators instead of ordinary lists. We could certainly go back to ordinary lists and hardly miss the extra byte or two of memory per notification. 😅

Reply:
Dependency dropped due to a refactor.